Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -84,12 +84,15 @@ namespace Terrain
 
     void TerrainMeshManager::SetConfiguration(const MeshConfiguration& config)
     {
-        if (m_config.CheckWouldRequireRebuild(config))
+        bool requireRebuild = m_config.CheckWouldRequireRebuild(config);
+
+        m_config = config;
+
+        if (requireRebuild)
         {
             m_rebuildSectors = true;
             OnTerrainDataChanged(AZ::Aabb::CreateNull(), TerrainDataChangedMask::HeightData);
         }
-        m_config = config;
 
```

Comment:
The issue here was that the OnTerrainDataChanged() call uses m_config to rebuild the sectors, so m_config needs to get set with the new values *before* calling OnTerrainDataChanged().